### DISCUSSION TITLE
Sherlock showing 'string indices' error

### DISCUSSION THREAD

**Discussion by charredsteak6** [sherlock-project/sherlock#2226] [Q&A] [answered]
![2024-07-09 10_25_09-Window](https://github.com/sherlock-project/sherlock/assets/134768335/67a59219-d109-4e37-b7fb-0810f9c18677)
I've installed sherlock using the command `apt install sherlock` and when i try to run `sherlock user123` I receive this error message. Am I doing something wrong?

**Comment by ppfeister** [comment]
Hey @charredsteak6 

Good catch. The Kali package seems to be a little bit further behind than I thought. I can't give an exact timeline on when an update will be available since it's community maintained, but I heard from the packager earlier today and it sounds like it'll be out this week.

I've made some changes here to better support those out of date packages. Can you check what happens now?

**Reply by charredsteak6** [reply]
![image](https://github.com/sherlock-project/sherlock/assets/134768335/acfc1908-ccb4-4117-987a-ae726ae75c9a)
Still seems to be giving me that error message. It's telling me that the version number is 0.14.3 and that 0.15.0 is available, however I have already installed that update through the git clone command. Not sure if that has anything to do with it, I'm kinda new at this haha.

**Reply by ppfeister** [reply, marked as answer]
Oh wow, 0.14.3 is much older. That answers my other confused question.

The version check failing shouldn't have prevented you from running Sherlock, but since you were on 0.14.3, it wasn't able to parse the new manifest either. Sherlock was adapted to accept some new keys in 0.14.4.

The good news is that 0.14.4 is actually available on Kali.

```bash
sudo apt update
sudo apt install --only-upgrade sherlock
```
Omit `--only-upgrade` if you want to update everything else while you're at it.

The update notification will still be there until the deb/kali packager pushes out an update later this week, but once you get on 0.14.4 it'll work as otherwise expected.

___

PS --- git clone isn't necessary when you're installing via apt/pip/dnf/etc. only when you're installing from source or making contributions.

**Reply by charredsteak6** [reply]
Ok thank you for the help I’ll give it a go later today .

**Reply by charredsteak6** [reply]
> Oh wow, 0.14.3 is much older. That answers my other confused question.
> 
> The version check failing shouldn't have prevented you from running Sherlock, but since you were on 0.14.3, it wasn't able to parse the new manifest either. Sherlock was adapted to accept some new keys in 0.14.4.
> 
> The good news is that 0.14.4 is actually available on Kali.
> 
> ```shell
> sudo apt update
> sudo apt install --only-upgrade sherlock
> ```
> 
> Omit `--only-upgrade` if you want to update everything else while you're at it.
> 
> The update notification will still be there until the deb/kali packager pushes out an update later this week, but once you get on 0.14.4 it'll work as otherwise expected.
> 
> PS --- git clone isn't necessary when you're installing via apt/pip/dnf/etc. only when you're installing from source or making contributions.

Hey I gave it a go and now its working perfectly, in fact its actually telling me now that I'm using version 0.15.0. Thanks again
